java-frontend/src/main/java/org/sonar/java/model/JavaTree.java

```diff
@@ -129,7 +129,7 @@ public void setParent(Tree parent) {
    *
    * @throws java.lang.UnsupportedOperationException if {@link #isLeaf()} returns {@code true}
    */
-  protected abstract Iterable<Tree> children();
+  protected abstract List<Tree> children();
```

Comment:
There are some unit tests that add the result of this method into another list to do assertions. It worth modifying them to simplify a bit.